frontend/javascripts/oxalis/model/accessors/tool_accessor.js

```diff
@@ -0,0 +1,124 @@
+  }
+  if (isInMergerMode) {
+    return "Volume annotation is disabled while the merger mode is active.";
+  }
+  if (!isSegmentationVisibleForMag) {
+    return "Volume annotation is disabled since no segmentation data can be shown at the current magnification. Please adjust the zoom level.";
+  }
+  return "Volume annotation is currently disabled.";
+};
+
+// eslint-disable-next-line import/prefer-default-export
+export function getDisabledInfoForTools(state: OxalisState) {
+  const isInMergerMode = state.temporaryConfiguration.isMergerModeEnabled;
+  const isZoomInvalidForTracing = isMagRestrictionViolated(state);
+  const maybeResolutionWithZoomStep = getRenderableResolutionForSegmentation(state);
+  const labeledResolution =
+    maybeResolutionWithZoomStep != null ? maybeResolutionWithZoomStep.resolution : null;
+  const isSegmentationVisibleForMag = labeledResolution != null;
+
+  const hasVolume = state.tracing.volume != null;
+  const hasSkeleton = state.tracing.skeleton != null;
+  const isSegmentationActivated = (() => {
+    const segmentationLayer = getSegmentationLayer(state.dataset);
+    if (segmentationLayer == null) {
+      return false;
+    }
+    return isLayerVisible(
+      state.dataset,
+      segmentationLayer.name,
+      state.datasetConfiguration,
+      state.temporaryConfiguration.viewMode,
+    );
+  })();
+
+  const genericDisabledExplanation = getExplanationForDisabledVolume(
+    isSegmentationActivated,
+    isInMergerMode,
+    isSegmentationVisibleForMag,
+    isZoomInvalidForTracing,
+  );
+
+  const disabledSkeletonExplanation =
+    "This annotation does not have a skeleton. Please convert it to a hybrid annotation.";
+
+  if (!hasVolume || !isSegmentationActivated || !isSegmentationVisibleForMag || isInMergerMode) {
+    // All segmentation-related tools are disabled.
+    const disabledInfo = {
+      isDisabled: true,
+      explanation: genericDisabledExplanation,
+    };
+    return {
+      [AnnotationToolEnum.MOVE]: {
+        isDisabled: false,
+        explanation: "",
+      },
+      [AnnotationToolEnum.SKELETON]: {
+        isDisabled: !hasSkeleton,
+        explanation: disabledSkeletonExplanation,
+      },
+      [AnnotationToolEnum.BRUSH]: disabledInfo,
+      [AnnotationToolEnum.TRACE]: disabledInfo,
+      [AnnotationToolEnum.FILL_CELL]: disabledInfo,
+      [AnnotationToolEnum.PICK_CELL]: disabledInfo,
+    };
+  }
+
+  const isZoomStepTooHighForBrushing = isZoomStepTooHighFor(state, AnnotationToolEnum.BRUSH);
+  const isZoomStepTooHighForTracing = isZoomStepTooHighFor(state, AnnotationToolEnum.TRACE);
+  const isZoomStepTooHighForFilling = isZoomStepTooHighFor(state, AnnotationToolEnum.FILL_CELL);
+
+  return {
+    [AnnotationToolEnum.MOVE]: {
+      isDisabled: false,
+      explanation: "",
+    },
+    [AnnotationToolEnum.SKELETON]: {
+      isDisabled: !hasSkeleton,
```

Comment:
Does this mean the skeleton tool is shown (but disabled) in volume-only tracings, whereas the volume tools are not shown in skeleton-only tracings?

Reply:
With #5422 this becomes a bit cleaner. Then, the logic is quite simple: skeleton-tools are shown if there is a skeleton and volume tools are shown if there's a volume annotation. The move-tool is always shown obviously.
The `isDisabled` property is still computed here, since that  information is used when cycling the current tool.